utils/merge.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Merge will return a new struct/map/slice of the same type as base and patch, with patch merged into base.
+// Specifically, patch's values will be preferred except when patch's value is `nil`.
+// Note: a referenced value (eg. *bool) will only be `nil` if the pointer is nil. If the value is a zero value,
+//       then that is considered a legitimate value. Eg, *bool(false) will overwrite *bool(true).
+//
+// Restrictions/guarantees:
+//   - base and patch will not be modified
+//   - base and patch can be pointers or values
+//   - base and patch must be the same type
+//   - if slices are different, this rule applies:
+//       - if patch is not nil, overwrite the base slice.
+//       - otherwise, keep the base slice
+//   - if maps are different, they will be merged according to the following rules:
```

Comment:
I think this documentation block needs to be updated to reflect the simpler semantics.

Reply:
Done. I had updated the one below, but missed this, thanks!